lib/cublas/linalg.jl

```diff
@@ -24,6 +24,11 @@ function LinearAlgebra.dot(x::StridedCuArray{T}, y::StridedCuArray{T}) where T<:
     dotc(n, x, y)
 end
 
+LinearAlgebra.dot(x::StridedCuArray{Bool}, y::StridedCuArray{T}) where {T<:Union{Float16, CublasReal, ComplexF16, CublasComplex}} = LinearAlgebra.dot(y, x)
+LinearAlgebra.dot(x::StridedCuArray{T}, y::StridedCuArray{Bool}) where {T<:Union{Float16, CublasReal, ComplexF16, CublasComplex}} = sum(view(x, y))
+LinearAlgebra.dot(x::StridedCuArray{<:Integer}, y::StridedCuArray{T}) where {T<:Union{Float16, CublasReal, ComplexF16, CublasComplex}} = LinearAlgebra.dot(y, x)
+LinearAlgebra.dot(x::StridedCuArray{T}, y::StridedCuArray{<:Integer}) where {T<:Union{Float16, CublasReal, ComplexF16, CublasComplex}} = LinearAlgebra.dot(x, convert(typeof(x), y))
```

Comment:
Not sure whether this is the best approach to do so or not. Maybe we can write a kernel instead, but does it worth for arrays of Float16

Reply:
The conversion may also fail for specific inputs. Since CUBLAS doesn't really support `dot` with arrays of Bool, I'm not  sure we want these definitions, but think we want a custom kernel instead.